### DISCUSSION TITLE
'MODULE_NOT_FOUND'

### DISCUSSION THREAD

**Discussion by AmrutaDeshpande21** [eclipse-theia/theia#11791] [Q&A] [answered]
I am facing to many errors in building Theia example application: https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#windows
Below is the error 

```
PS D:\CeliumDevices\theia-hello-world-extension\my-app\theia> yarn browser build
yarn run v1.22.19
$ yarn -s --cwd examples/browser build
node:internal/modules/cjs/loader:988
  throw err;
  ^

    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15)
    at Function.Module._load (node:internal/modules/cjs/loader:833:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:22:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
    at node:internal/main/run_main_module:22:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Any help to debug would be highly appreciated

**Comment by msujew** [comment]
Hey @AmrutaDeshpande21,

Does `yarn install` successfully execute in the repository? Also which version of `node` are you using?

There seems to be something wrong with the initial setup, I haven't ever seen that specific error personally.

**Reply by AmrutaDeshpande21** [reply]
yarn 1.22.19
node v16.18.0

**Reply by vince-fugnitto** [reply]
@AmrutaDeshpande21 are you attempting to build and run the `theia` repo (https://github.com/eclipse-theia/theia) or the theia-extension generator (`hello-world` template)?

In any case, can you provide the steps you executed in order to build?

**Reply by AmrutaDeshpande21** [reply]
@vince-fugnitto I am attempting Theia repo
I installed the prerequisites and followed the document https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#windows to build an example application. However I was stuck in "Build core, extensions and examples packages " this part of it. Where it says to build the application.

**Reply by vince-fugnitto** [reply]
@AmrutaDeshpande21 have you seen and tried https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#building-on-windows?

**Reply by AmrutaDeshpande21** [reply]
yes I have tried this. Do I need to install the windows-build-tools ?

**Reply by vince-fugnitto** [reply]
> yes I have tried this. Do I need to install the windows-build-tools ?

@AmrutaDeshpande21 I don't think its necessary anymore with newer node versions, but unfortunately I'm not on Windows myself.

Can you do a fresh build, and paste the full error or output:
From the root of the repository:
- `git clean -ffdx`
- `yarn`
- `yarn browser build`

**Reply by AmrutaDeshpande21** [reply]
yes running the command in git. waiting for it to complete.

**Reply by AmrutaDeshpande21** [reply]
![image](https://user-images.githubusercontent.com/91452767/197175428-129b8a4b-f9c6-4029-bdfa-c056ca0105a2.png)

This is taking really long time

**Reply by AmrutaDeshpande21** [reply]
@vince-fugnitto could you please look into this?

**Reply by vince-fugnitto** [reply]
@AmrutaDeshpande21 it looks like the command `yarn browser start` works (starting the browser backend). You can now simply connect to it by opening http://localhost:3000/ in the browser.

**Reply by AmrutaDeshpande21** [reply]
Thank you I was able to connect to the server.

**Reply by AmrutaDeshpande21** [reply]
Earlier I was trying to run the yarn commands in Theia Blueprint terminal and I got Module not found error. But worked with git.

**Comment by vince-fugnitto** [comment, marked as answer]
I believe the discussion is resolved.
